src/__tests__/connection/database/index.test.js

```diff
@@ -1,60 +1,101 @@
+    describe('for database connection without username required', () => {
+      const model = getModel('user@contoso.com', null, false);
 
-    it('should get the email', () => {
-      expect(databaseUsernameValue(model)).toEqual('user@contoso.com');
+      it('should get the email', () => {
+        expect(databaseUsernameValue(model)).toEqual('user@contoso.com');
+      });
     });
-  });
 
-  describe('for database connection with username required', () => {
-    const model = getModel('user@contoso.com', 'user', true);
+    describe('for database connection with username required', () => {
+      const model = getModel('user@contoso.com', 'user', true);
 
-    it('should get the username when `emailFirst` is not set', () => {
-      expect(databaseUsernameValue(model)).toEqual('user');
-    });
-    it('should get the username when `emailFirst` is false', () => {
-      expect(databaseUsernameValue(model, { emailFirst: false })).toEqual('user');
-    });
-    it('should get the email when `emailFirst` is true', () => {
-      expect(databaseUsernameValue(model, { emailFirst: true })).toEqual('user@contoso.com');
-    });
+      it('should get the username when `emailFirst` is not set', () => {
+        expect(databaseUsernameValue(model)).toEqual('user');
+      });
+      it('should get the username when `emailFirst` is false', () => {
+        expect(databaseUsernameValue(model, { emailFirst: false })).toEqual('user');
+      });
+      it('should get the email when `emailFirst` is true', () => {
+        expect(databaseUsernameValue(model, { emailFirst: true })).toEqual('user@contoso.com');
+      });
 
-    describe('and only email address is filled in', () => {
-      const model = getModel('user@contoso.com', null, true);
+      describe('and only email address is filled in', () => {
+        const model = getModel('user@contoso.com', null, true);
 
-      it('should get the email address', () => {
-        expect(databaseUsernameValue(model)).toEqual('user@contoso.com');
+        it('should get the email address', () => {
+          expect(databaseUsernameValue(model)).toEqual('user@contoso.com');
+        });
+      });
+    });
+  });
+  describe('initDatabase', () => {
+    describe('calls initNS with the correct additionalSignUpFields', () => {
+      describe('with a valid hidden field', () => {
+        const model = Immutable.fromJS({});
+        const modelOut = initDatabase(model, {
+          additionalSignUpFields: [
+            {
+              type: 'hidden',
+              name: 'hidden_field',
+              value: 'hidden_value'
+            }
+          ]
+        });
+        const modelOutJS = modelOut.toJS();
+        expect(modelOutJS.field).toEqual({
+          hidden_field: { showInvalid: false, valid: true, value: 'hidden_value' }
+        });
+        expect(modelOutJS.database.additionalSignUpFields).toEqual([
+          {
+            name: 'hidden_field',
```

Comment:
Super-minor but this would look nicer if the values were in same order each time:

```
{
  "type" : "hidden",
  "name" : "hidden_field",
  "value" : "hidden_value",
}
```